lib/sync/syncer.go

```diff
@@ -267,18 +244,21 @@ func (s *Syncer) work(height uint64, nodeAddrs []string) bool {
 				begin := time.Now()
 				syncInfo, err = s.fetcher.Fetch(ctx, syncInfo)
 				if err != nil {
-					if err != context.Canceled {
-						s.logger.Error("fetch failure", "err", err, "height", height)
+					if err == context.Canceled {
+						break L
 					}
+					s.logger.Error("fetch failure", "err", err, "height", height)
+					continue
```

Comment:
How about adding sleep here?
In our environment, to occur this error means same error can be occurred again by a block from any other nodes.
Because the last block is invalid.
So it makes tons of error log messages even only 30 seconds.

Reply:
No, it is not related to check prev block interval. Dont display err logs every 30 min